### PR TITLE
gossipsub v1.1 draft r2

### PR DESCRIPTION
**Pull request by vyzo**
Updates the gossipsub v1.1 spec to document/specify new behaviours introduced and tested in the reference implementation.

### CODE REVIEW

File: pubsub/gossipsub/gossipsub-v1.1.md [hunk 1]
```diff
@@ -58,6 +60,22 @@ as the protocol string.
 
 ## Protocol extensions
 
+### Direct Peering Agreements
```
Comment (daviddias) [inline pubsub/gossipsub/gossipsub-v1.1.md:63]:
```suggestion
### Explicit Peering Agreements
```

When preparing the presentation for the Matrix, I found it simpler to explain by calling these "Explicit" to contrast with the "Peering Agreements" that are created through GRAFTing

Comment (vyzo) [reply pubsub/gossipsub/gossipsub-v1.1.md:63]: Well, the explicit is slightly different -- they are direct peers in the protocol, hence direct peering agreements.

Comment (daviddias) [reply pubsub/gossipsub/gossipsub-v1.1.md:63]: But they are set by the user/app, hence explicit.

All the peering agreements are always between two nodes, hence one could call all of them direct.

Comment (vyzo) [reply pubsub/gossipsub/gossipsub-v1.1.md:63]: Precisely! These are agreements between pairs of peers, so they are called direct peering agreements in the protocol.

Comment (vyzo) [reply pubsub/gossipsub/gossipsub-v1.1.md:63]: So the contrast is between "Mesh peering agreements" and "direct peering agreements".

Comment (vyzo) [reply pubsub/gossipsub/gossipsub-v1.1.md:63]: At any rate I don't feel all that strongly about the wording in the section title; if you insist I will bulge :)

Comment (daviddias) [reply pubsub/gossipsub/gossipsub-v1.1.md]: we resolved async, discussed the merits of both terms and ended up with "explicit. Next commit updates the spec, including doctoc :)

**Pull request merged**